compiler/src/iree/compiler/Codegen/Common/test/optimize_tensor_insert_extract_slices.mlir

```diff
@@ -138,3 +138,53 @@ func.func @_batch_matmul_narrow_n_2_dispatch_4_unpack_i32() attributes {translat
 // CHECK: %[[EMPTY:[a-zA-Z0-9]+]] = tensor.empty() : tensor<3x2xi32>
 // CHECK: %[[TRANS:[a-zA-Z0-9]+]] = vector.transpose %5, [1, 0] : vector<2x8xi32> to vector<8x2xi32>
 // CHECK: vector.transfer_write %[[TRANS]], %[[EMPTY]][%c0, %c0] {in_bounds = [false, true]} : vector<8x2xi32>, tensor<3x2xi32>
+
+// -----
+
+func.func @subset_hoisting_invariant_tensor(%init: tensor<64x64xf32>, %t: tensor<64x64xf32>) -> tensor<64x64xf32> {
+  %c0 = arith.constant 0 : index
+  %c1 = arith.constant 1 : index
+  %c8 = arith.constant 8 : index
+  %empty = tensor.empty() : tensor<8x1xf32>
+  %loop = scf.for %i = %c0 to %c8 step %c1 iter_args(%arg1 = %init) -> tensor<64x64xf32> {
+    %slice1 = tensor.extract_slice %arg1[1, 0][8, 1][1, 1] : tensor<64x64xf32> to tensor<8x1xf32>
```

Comment:
I am confused as to why this is legal. You are reading from the value that was updated in the previous iteration of the loop. How is that hoistable?

Reply:
The hoisting only happens if we can provide that the slice that we are reading, is the one that was updated in the last iteration. So we directly forward the value being written to the slice and use it directly.

Reply:
I updated the documentation to explain why this transformation is legal